verifier/src/quote.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+use crate::{Accessor, VerificationMessage, VerificationOutput, Verifier};
+use core::fmt::Formatter;
+use mc_sgx_dcap_types::Quote3;
+use p256::ecdsa::VerifyingKey;
+
+/// Verifier for ensuring a quote was signed with the provided key
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Quote3Verifier<T> {
+    key: VerifyingKey,
+    _phantom: core::marker::PhantomData<T>,
+}
+
+impl<T> Quote3Verifier<T> {
+    /// Create a new instance.
+    ///
+    /// The `key` should be retrieved from the leaf certificate of the quote's
+    /// [`CertificationData`].
+    pub fn new(key: VerifyingKey) -> Self {
+        Self {
+            key,
+            _phantom: core::marker::PhantomData,
+        }
+    }
+}
+
+impl<T: AsRef<[u8]>, E: Accessor<Quote3<T>>> Verifier<E> for Quote3Verifier<T> {
+    type Value = ();
+    fn verify(&self, evidence: &E) -> VerificationOutput<Self::Value> {
+        let quote = evidence.get();
+        let is_success = quote.verify(&self.key).is_ok() as u8;
+
+        VerificationOutput::new((), is_success.into())
+    }
+}
+
+impl<T> VerificationMessage<()> for Quote3Verifier<T> {
+    fn fmt_padded(
+        &self,
+        f: &mut Formatter<'_>,
+        pad: usize,
+        result: &VerificationOutput<()>,
+    ) -> core::fmt::Result {
+        let message = if result.is_success().into() {
+            "The quote was signed with the provided key"
+        } else {
+            "The quote signature did not match provided key"
+        };
```

Comment:
up for suggestions on this display
The other verifiers show the actual and expected values, but showing the signature or public key doesn't seem like it's going to help the client of the code.
Showing the entire quote bytes doesn't seem like it will help much either.

I'm guessing this shouldn't fail very often since the key itself is on the tail of the quote